manifests/config/apache.pp

```diff
@@ -112,10 +109,9 @@
   Optional[String] $user = undef,
   Optional[Stdlib::HTTPUrl] $foreman_url = undef,
   Optional[String] $access_log_format = undef,
-  Boolean $ipa_authentication = false,
   Hash[String, Any] $http_vhost_options = {},
   Hash[String, Any] $https_vhost_options = {},
-  Boolean $keycloak = false,
+  Enum['internal', 'ipa', 'ipa_with_api', 'keycloak'] $authentication = 'internal',
```

Comment:
I'd rather have `external_authentication` as the key with undefined, `ipa`, `ipa_with_api` and `keycloak` as options since they're all mutually exclusive, while `internal` is always there.

Reply:
Good feedback, wasn't quite sure about it.